hypothesis-python/tests/cover/test_filter_rewriting.py

```diff
@@ -343,14 +343,23 @@ def test_warns_on_suspicious_string_methods(method):
     assert fs.min_size == 1
 
 
-@pytest.mark.parametrize("method", [str.isidentifier, str.isalnum])
+@pytest.mark.parametrize("method", [str.isalnum])
 def test_bumps_min_size_and_filters_for_content_str_methods(method):
     s = unwrap_strategies(st.text())
     fs = s.filter(method)
     assert fs.filtered_strategy.min_size == 1
     assert fs.flat_conditions == (method,)
 
 
+@pytest.mark.parametrize("al", [["c", "d", "e", "f"], "cdef"])
+def test_filter_alphabet_properly_rewritten(al):
+    @given(st.text(alphabet=al).filter(str.isidentifier))
+    def wrapped(xs):
+        assert set(xs).issubset(set("cdef"))
```

Comment:
Conceptually we want to `assert set(xs).issubset(al)`, since we're enforcing a filter, and we also want to `assert xs.isidentifier()`!  

Checking for a much wider range of input alphabets would be encouraging too (e.g. non-ascii, not guaranteed to be identifiers, ...), and as a style thing I'd do this using `st.data()` instead of nested functions - it's the same semantics.

Finally, it'd be nice to include some tests for what happens if the filter is unsatisfiable - I think the intervals logic will let us have a nice error message for this.